### PR TITLE
build: fix linking when used in several translation units

### PR DESCRIPTION
**Pull request by drewdzzz**
All non-template methods defined outside classes are not implicitly `inline` - that violates one definition rule when the connector is included in several translaction units. Let's mark such methods as `inline` to fix this issue.

### CODE REVIEW

File: test/LinkTest/Main.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2010-2020, Tarantool AUTHORS, please see AUTHORS file.
```
Comment (mkostoevr) [inline test/LinkTest/Main.cpp:2]: *`Copyright 2010-2024`

File: test/LinkTest/Secondary.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2020, Tarantool AUTHORS, please see AUTHORS file.
```
Comment (mkostoevr) [inline test/LinkTest/Secondary.cpp:2]: Ditto.

**Review by mkostoevr — APPROVED** [review]

**Review by CuriousGeorgiy — APPROVED** [review]

File: test/LinkTest/Main.cpp (hunk 1 above)
Comment (drewdzzz) [reply test/LinkTest/Main.cpp]: Fixed. Also, added missing copyright to the header.

File: test/LinkTest/Secondary.cpp (hunk 2 above)
Comment (drewdzzz) [reply test/LinkTest/Secondary.cpp]: Fixed.

**Pull request merged**